Admin Scholarship Dashboard View

BrodrickS opened the pull request:
For Task: https://trello.com/c/WsCKgsvF

BrodrickS committed:
Initial changes for task

BrodrickS committed:
Merge branch 'Admin-ScholarshipDashboard-old' into admin-scholarship-dashboard

# Conflicts:
#	frontend/src/Admin/Page.js
#	frontend/src/Admin/queries/portfolioPeriods.graphql

JohnPMurray commented on frontend/src/Admin/components/NavBar.js:
```diff
@@ -49,6 +49,32 @@ export default class NavBar extends Component {
             <Nav className='ml-auto' navbar>
+              <NavItem>
+                <NavLink tag={Link} to='/'>
+                  Dashboard
```
I would say change this to shows dashboard or something

BrodrickS replied:
Changed to "Shows Dashboard"

JohnPMurray commented on frontend/src/Admin/components/PortfolioCard.js:
```diff
@@ -0,0 +1,234 @@
+import React, { Component, Fragment } from 'react'
```
How much of this file was copied from shows card? Is there a way we can use any shared components?
For the student version of this I ended up turning some of the matching internals into components that they both used.

BrodrickS replied:
Just got it working. Most of it is duplicated. The "shows" card has some additional functionality, I'll take a look at reusing components.

BrodrickS committed:
Changes for consistancy

Used to be "show/new" now "portfolio/new"
Changed all instances of path "portfolios" (plural) to "portfolio" (singular)

johnbyrne727rit commented on frontend/src/Admin/queries/portfolioPeriod.graphql:
```diff
@@ -0,0 +1,71 @@
+query PortfolioPeriod($id: ID!) {
+  show(id: $id) {
```
Is this pulling shows from the BE?

johnbyrne727rit replied:
Discussed irl, should be portfolioPeriod now.

JohnPMurray reviewed (changes requested):
Make sure to actually pull portfolio periods. Also I made a few comments to adress

JohnPMurray commented:
I know you didn't make the create a portfolio period page, but we should change it to link to the portfolios dashboard (rather than the shows dashboard) when its done

BrodrickS committed:
Changes for consistancy

Changed all instances of "portfolio" (like a submission) to "portfolioPeriod" (like a show)

BrodrickS committed:
Consistency - New Portfolio Period

Once the new Portfolio Period has been created then redirect is back to the portfolio dashboard

BrodrickS commented:
> I know you didn't make the create a portfolio period page, but we should change it to link to the portfolios dashboard (rather than the shows dashboard) when its done

Done

BrodrickS committed:
Consistency - Changed Admin Dashboard Name

"Dashboard" changed to "Shows Dashboard"

BrodrickS committed:
Portfolio Period Cards Added

Now pulls data from the database and shows the information in a card. The card data still needs to be finalized.

BrodrickS committed:
Removed Unused "PortfolioPeriod.graphql" Query

File was unused, and a duplicate of "show.graphql"

BrodrickS committed:
Stats Added to Admin PortfolioPeriod Card

NOTE: Blocked until "Add Portfolio Submission" PR goes through. See TODO note for task
Added a "Total Portfolios" count
Added a "Submitted" count
Added a "Pending" count

BrodrickS committed:
Merge branches 'admin-scholarship-dashboard' and 'dev' of https://github.com/johnbyrne727rit/gallery-gateway into admin-scholarship-dashboard

BrodrickS committed:
Continued - Stats Added to Admin PortfolioPeriod Card

Added a new Resolver Type 'PortfolioPeriodType' in resolvers/types
    This creates a new property 'portfolios' that includes all portfolios created for this portfolio period.
Added this new type to index.js
Added new property to the frontend admin portfolioperiod query
Added new data to PortfolioPeriodCard.js

johnbyrne727rit reviewed (approved):
Looks groovy to me

JohnPMurray reviewed (approved):
looks good
